### PR TITLE
EVPN host-flap settings

### LINKED ISSUE #914: Support for EVPN host-flap detection
**Issue Type**

- [x] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

I would like to change the EVPN host-flap detection default behavior which is currently set to window 180.0 and threshold 5.

**Describe the solution you'd like**

Make it possible to change EVPN host-flap detection within the eos_designs role.

**Describe alternatives you've considered**

It is possible to use the eos_cli_config_gen role for BGP configuration with address_family_evpn support. However host-flap detection is also not supported there nor I'm not sure I could stitch both roles (eos_designs + eos_cli_config_gen) together.

### PR DESCRIPTION
**Pull request by mpergament**
## Change Summary

EVPN host-flap settings

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #914 

## Component(s) name

`arista.avd.eos_designs`
`arista.avd.eos_cli_config_gen`

## Proposed changes
```YML
evpn_hostflap_detection:
  window: 20
  threshold: 30
```

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/overlay/ebgp/router-bgp.j2 [hunk 1]
```diff
@@ -23,6 +23,11 @@ router_bgp:
         activate: false
 
   address_family_evpn:
+{%     if switch.evpn_role != "server" and evpn_hostflap_detection.window is arista.avd.defined and evpn_hostflap_detection.threshold is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/overlay/ebgp/router-bgp.j2]: Why not on a server? Remember that server role can be on a leaf where we services are also deployed. Maybe use `if switch.configure_tenants is arista.avd.defined(true)` or `if switch.vtep_ip is arista.avd.defined` ?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 2]
```diff
@@ -154,6 +154,12 @@
 {%     endif %}
 
 ### Router BGP EVPN Address Family
+{%     if evpn_hostflap_detection.window is arista.avd.defined and evpn_hostflap_detection.threshold is arista.avd.defined %}
+#### EVPN Host Flapping Settings
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2:159]: Please add an empty line between line 157 and 158. To give a consistent MD output.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/overlay/ebgp/router-bgp.j2 [hunk 3]
```diff
@@ -23,6 +23,20 @@ router_bgp:
...
+{%     if switch.vtep_ip is arista.avd.defined and ( evpn_hostflap_detection.window is arista.avd.defined or evpn_hostflap_detection.threshold is arista.avd.defined or evpn_hostflap_detection.enabled is arista.avd.defined(false)) %}
+    evpn_hostflap_detection:
+{%         if evpn_hostflap_detection.window is arista.avd.defined %}
+      window: {{ evpn_hostflap_detection.window }}
+{%         endif %}
+{%         if evpn_hostflap_detection.threshold is arista.avd.defined %}
+      threshold: {{ evpn_hostflap_detection.threshold }}
+{%         endif %}
+{%         if evpn_hostflap_detection.enabled is arista.avd.defined(false) %}
+      enabled: 'false'
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/overlay/ebgp/router-bgp.j2]: Remove quotes. Maybe even refactor so line 34-38 could be replaced with `enabled: {{ evpn_hostflap_detection.enabled | arista.avd.default(true) }}

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Current tests are not working with the quotes around 'false'.

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/documentation/devices/DC1-SPINE1.md [hunk 4]
```diff
@@ -447,6 +447,11 @@ ip route vrf MGMT 0.0.0.0/0 192.168.200.5
 
 ### Router BGP EVPN Address Family
 
+#### EVPN Host Flapping Settings
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/documentation/devices/DC1-SPINE1.md]: Should not be generated for devices where it is not configured.
Problem is that documentation looks at wrong variables.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 5]
```diff
@@ -154,6 +154,25 @@
 {%     endif %}
 
 ### Router BGP EVPN Address Family
+{%     if evpn_hostflap_detection.window is arista.avd.defined or evpn_hostflap_detection.threshold is arista.avd.defined or  evpn_hostflap_detection.enabled is arista.avd.defined(false) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]: device documentation should only look at structured_config variables. So we need to adjust this documentation to use the same vars as you use in the eos config template. I can help if you like.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
We need to redo the documentation. Let me know if I can help.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGMT!

**Pull request merged**